pre_commit_hooks/string_fixer.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import absolute_import
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import argparse
+import re
+import tokenize
+
+
+double_quote_starts = tuple(s for s in tokenize.single_quoted if '"' in s)
+
+
+def handle_match(m):
+    string = m.group(0)
+
+    for double_quote_start in double_quote_starts:
+        if string.startswith(double_quote_start):
+            meat = string[len(double_quote_start):-1]
+            if '"' in meat or "'" in meat:
+                break
+            return (
+                double_quote_start.replace('"', "'") +
+                string[len(double_quote_start):-1] +
+                "'"
+            )
+    return string
+
+
+def fix_strings(filename):
+    return_value = 0
+
+    lines = []
+
+    with open(filename, 'r') as read_handle:
+        for line in read_handle:
+            if '"""' in line:
+                # Docstrings are hard, fuck it
+                lines.append(line)
+            else:
+                result = re.sub(tokenize.String, handle_match, line)
+                lines.append(result)
+                return_value |= int(result != line)
+
+    with open(filename, 'w') as write_handle:
+        for line in lines:
+            write_handle.write(line)
```

Comment:
Token considers """ Foo """ to be a quoted string. Docstring test fails if I remove the special case.